CPAC/QPP/prep_QPP.py

```diff
@@ -0,0 +1,548 @@
+    exts = ['.'+ ext.lstrip('.')for ext in exts]
+    output_dict_list={}
+    for root, _,files in os.walk(pipeline_output_folder):
+        for filename in files:
+            filepath=os.path.join(root,filename)
+
+            if not any(fnmatch.fnmatch(filepath,pattern)for pattern in nifti_globs):
+                continue
+            if not any(filepath.endswith(ext)for ext in exts):
+                continue
+            relative_filepath=filepath.split(pipeline_output_folder)[1]
+            filepath_pieces=filter(None, relative_filepath.split("/"))
+            resource_id = filepath_pieces[1]
+
+            if resource_id not in search_dir:
+                continue
+
+            scan_id_string = filepath_pieces[2]
+            strat_info = "_".join(filepath_pieces[3:])[:-len(ext)]
+            unique_resource_id=(resource_id,strat_info)
+            if unique_resource_id not in output_dict_list.keys():
+
+                output_dict_list[unique_resource_id] = []
+
+            unique_id = filepath_pieces[0]
+
+            scan_id = scan_id_string.replace("_scan_", "")
+            scan_id = scan_id.replace("_rest", "")
+
+            new_row_dict = {}
+            new_row_dict["participant_session_id"] = unique_id
+            new_row_dict["participant_id"], new_row_dict["Sessions"] = \
+                unique_id.split('_')
+
+            new_row_dict["Scan"] = scan_id
+            new_row_dict["Filepath"] = filepath
+
+            print('{0} - {1} - {2}'.format(unique_id, scan_id,
+                                           resource_id))
+            output_dict_list[unique_resource_id].append(new_row_dict)
+
+        #analysis, grouped either by sessions or scans.
+
+    return output_dict_list
+
+
+def create_output_df_dict(output_dict_list,inclusion_list):
+
+    import pandas as pd
+
+    output_df_dict={}
+
+
+    for unique_resource_id in output_dict_list.keys():
+
+        ##This dataframe will give you what is in the C-PAC output directory for individual level analysis outputs##
+        new_df = pd.DataFrame(output_dict_list[unique_resource_id])
+
+        col_names = new_df.columns.tolist()
+
+        if inclusion_list:
+            #this is for participants only, not scans/sessions/etc
+
+            new_df=new_df[new_df.participant_id.isin(inclusion_list)]
+
+        if new_df.empty:
+
+                print("No outputs found for {0} the participants "\
+                      "listed in the group manalysis participant list you "\
+                      "used. Skipping generating model for this "\
+                      "output.".format(unique_resource_id))
+                continue
+
+        if unique_resource_id not in output_df_dict.keys():
+                output_df_dict[unique_resource_id] = new_df
+
+    return output_df_dict
+
+
+def gather_outputs(pipeline_folder,resource_list,inclusion_list):
+    nifti_globs,search_dir = gather_nifti_globs(pipeline_folder,resource_list,derivatives=None)
```

Comment:
And this.

Reply:
Do you mean excess white spaces here? or is it any other issue?

Reply:
Sorry- I was referring to the other two comments above this one, for this file. Pasted here:

What is different here compared to the original gather_nifti_globs in cpac_group_runner? Should combine functionality into the original one.